internal/resolvers/imgur/resolver.go

```diff
@@ -27,7 +29,19 @@ type Resolver struct {
 }
 
 func (r *Resolver) Check(ctx context.Context, url *url.URL) (context.Context, bool) {
-	return ctx, utils.IsSubdomainOf(url, "imgur.com")
+	for _, domain := range VALID_IMGUR_DOMAINS {
+		result := utils.IsSubdomainOf(url, domain)
+		if result {
+			return nil, result
+		}
+
+		result = utils.IsDomain(url, domain)
+		if result {
+			return nil, result
+		}
+	}
+
+	return ctx, false
```

Comment:
This seems to be causing the error you're running into. The `Context` returned by the method is later used in `internal/resolvers/default/link_resolver.go` and will cause an error if it's `nil`. The error can be resolved with this change:

```suggestion
	for _, domain := range VALID_IMGUR_DOMAINS {
		result := utils.IsSubdomainOf(url, domain)
		if result {
			return ctx, result
		}

		result = utils.IsDomain(url, domain)
		if result {
			return ctx, result
		}
	}

	return ctx, false
```

---

However, even after making this change, imgur.io domains will not be resolved correctly. [It seems that `koffeinsource/go-imgur` does not support that domain yet](https://github.com/koffeinsource/go-imgur/blob/master/fromURL.go#L20-L44).

Reply:
Small update: I have created https://github.com/koffeinsource/go-imgur/pull/16 at the upstream repository to raise awareness.